code/Player/Grub/Controller/GrubController.cs

```diff
@@ -135,6 +145,24 @@ protected void UpdateRotation()
 		}
 	}
 
+	private void UpdateAirTime()
+	{
+		// Don't force resolve if we're already resolved or we are intentionally in the air (jetpack, parachute, long fall).
+		if ( Entity.Resolved || !Velocity.IsNearlyZero() )

```

Comment:
Can we up the tolerance on this a bit? There might be a few cases where there is a tiny bit of velocity left over. 1f should be fine here. Good to merge after that.